### PR TITLE
[MOS-263] Add HFP DTMF handling

### PR DESCRIPTION
**Pull request by SP2FET**
**Description**

Added handling of HFP DTMF code sending,
fixed DTMF sending in cellular's code

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible.
- [x] Has documentation updated

### CODE REVIEW

File: module-utils/utility/Utils.cpp [hunk 1]
```diff
@@ -61,6 +61,7 @@ namespace utils::filesystem
 
 namespace utils
 {
+
```
[the remark this answers is not in the document]
Comment (SP2FET) [reply module-utils/utility/Utils.cpp]: removed, thanks!

**Review by Lefucjusz — APPROVED** [review]

**Review by pawel-mudita — APPROVED** [review]

File: module-utils/utility/Utils.hpp [hunk 2]
```diff
@@ -23,6 +23,26 @@ namespace utils
     std::string bytesToHex(const std::vector<std::uint8_t> &bytes);
     std::vector<std::uint8_t> hexToBytes(const std::string &hex);
 
+    template <typename T> inline char singleDigitToChar(T digit)
+    {
+        static_assert(std::is_integral<T>::value, "Integral number required.");
```
Comment (pholat) [inline module-utils/utility/Utils.hpp:28]: isn't digit singular by definition?

File: module-utils/utility/Utils.hpp [hunk 3]
```diff
@@ -23,6 +23,26 @@ namespace utils
...
+            return '\0';
+        }
+        // converting digit to ASCII code
+        return static_cast<char>(digit + '0');
+    }
+
+    template <typename T> inline std::string singleDigitToString(T digit)
+    {
+        auto charDigit = singleDigitToChar(digit);
+        if (charDigit == '\0') {
```
Comment (pholat) [inline module-utils/utility/Utils.hpp:40]: why not throw? :) we have exceptions

File: module-services/service-cellular/ServiceCellular.cpp [hunk 4]
```diff
@@ -1474,7 +1474,8 @@ bool ServiceCellular::transmitDtmfTone(uint32_t digit)
     at::Result resp;
     if (channel) {
         auto command           = at::factory(at::AT::QLDTMF);
-        std::string dtmfString = "\"" + std::string(1, digit) + "\"";
+        std::string dtmfString = "\"" + utils::singleDigitToString(digit) + "\"";
```
Comment (pholat) [inline module-services/service-cellular/ServiceCellular.cpp:1477]: it's a shame that we do not have dedicated parser to this response

File: module-bluetooth/Bluetooth/interface/profiles/HFP/HFP.cpp [hunk 5]
```diff
@@ -316,10 +316,12 @@ namespace bluetooth
             // LOG_DEBUG("Attach number to voice tag. Sending '1234567\n");
             hfp_ag_send_phone_number_for_voice_tag(aclHandle, "1234567");
             break;
-        case HFP_SUBEVENT_TRANSMIT_DTMF_CODES:
-            LOG_DEBUG("Send DTMF Codes: '%s'\n", hfp_subevent_transmit_dtmf_codes_get_dtmf(event));
+        case HFP_SUBEVENT_TRANSMIT_DTMF_CODES: {
+            auto digitStr = hfp_subevent_transmit_dtmf_codes_get_dtmf(event);
+            LOG_DEBUG("Send DTMF Codes: '%s'\n", digitStr);
+            cellularInterface->sendDTMFCode(const_cast<sys::Service *>(ownerService), utils::toNumeric(digitStr));
```
Comment (pholat) [inline module-bluetooth/Bluetooth/interface/profiles/HFP/HFP.cpp:322]: in this pr: here we convert to numeric, to later on convert to string again? :)

File: module-utils/utility/Utils.hpp (hunk 2 above)
Comment (SP2FET) [reply module-utils/utility/Utils.hpp:28]: it should be, but there's no type for it

File: module-utils/utility/Utils.hpp (hunk 3 above)
Comment (SP2FET) [reply module-utils/utility/Utils.hpp:40]: rewritten to use exceptions

File: module-services/service-cellular/ServiceCellular.cpp (hunk 4 above)
Comment (SP2FET) [reply module-services/service-cellular/ServiceCellular.cpp:1477]: tbh I didn't want to save the world, only fix DTMF issue :P

**Pull request merged**